barman/backup_executor.py

```diff
@@ -1787,6 +1787,9 @@ def _pg_get_metadata(self, backup_info):
             summarize_wal = self.postgres.get_setting("summarize_wal")
             backup_info.set_attribute("summarize_wal", summarize_wal)
 
+        # Set total size of the PostgreSQL server
+        backup_info.set_attribute("cluster_size", self.postgres.current_size)
```

Comment:
`current_size` uses the total size of tablespaces.

```sql
postgres=# SELECT sum(pg_database_size(oid)) FROM pg_database;
    sum
-----------
 665407665
(1 row)
```

```sql
postgres=# SELECT sum(pg_tablespace_size(oid)) FROM pg_tablespace ;
    sum
-----------
 666034993
(1 row)
```

Who should we rely on, databases or tablespaces?

(Just joking).

Good catch on using that existing property.

Reply:
I found this property but also checked the pg_database_size! What i could find about differences between is that pg_tablespace_size also sums the size of global objects, that's why the size is greater using tablespaces!